.gitignore

```diff
@@ -1,10 +1,39 @@
+# Compiled Object files, Static and Dynamic libs (Shared Objects)
+*.o
+*.a
+*.so
+
+# Folders
+_obj
+_test
+
+# Architecture specific extensions/prefixes
+*.[568vq]
```

Comment:
Needs a rebase, and I'd say strip this down to what I merged in #1 unless you know you need to gitignore each of these lines.